subjects/corewar/README.md

```diff
@@ -0,0 +1,366 @@
+- `sti`: writes the value of the register passed as the first argument into to address obtained by summing the last two arguments. (the VM will apply `% IDX_MOD` to the obtained address)
+- `fork`: creates a new process which will be an exact (deep) copy of the current one except for the PC which will be at the address specified in the argument. (the VM will apply `% IDX_MOD` to the first argument)
+- `lld`, `lldi`, `lfork`: those instructions are the long versions of `ld`, `ldi` and `fork`, which means the addresses won't be truncated by `IDX_MOD`. (since those instructions are "long" the modulo operation won't be applied on them)
+- `nop`: this operation do nothing, it can still take one argument and has a pcode so be careful on how you implement it.
+
+> All addresses are relative to the current PC of the process.
+
+#### The `IDX_MOD` constant usage
+
+Some instructions have to truncate the addresses they are given by applying modulo `IDX_MOD`.
+This feature prevent players from reaching spaces in memory that are too far from the current PC, which means processes won't be able to attack each other straight away but will need to move by doing smaller steps, this help having more balanced games.
+
+#### Pcode field
+
+Some instruction may accept different kind of parameters, and those parameters may require different sizes.
+For those instructions the assembler will use a byte after the opcode to inform the VM which kind of parameters to expect.
+
+Every two bits of this byte will inform the VM about the type of a parameter.
+
+- `01` will be used for a register parameter.
+- `10` will be used for a direct parameter.
+- `11` will be used for an indirect parameter.
+
+An example of a pcode byte could be `01111000`:
+
+Here the first parameter is expected to be a register, the second should be an indirect value and the third a direct value.
+
+> Those numbers are in binary base, for example `10` in binary is `2` in decimal.
+
+#### Has IDX field
+
+Some instruction will sum up direct values and/or use them as addresses to lookup into the arena memory space. For this reason we know that a full 32 bits integer won't be necessary and to save space we do save direct values on only two bytes in those cases.
+
+You have `Had Idx` column in the Instruction Set table, when it is true you must save and read direct parameters on two bytes, when false they will be saved on four bytes as usual.
+
+#### The Carry flag
+
+In all processors you will find a carry which is a flag that notify some edge case states to allow special operations, ensure security and serve many others functionalities.
+
+The Carry of the processes in the VM is in that sense a simplified version of this concept.
+
+The following commands will modify the carry of the process executing them: `ld`, `add`, `sub`, `and`, `or`, `xor`.
+
+Those commands will set the carry to `true` if the result of their respective operations is `0`, and set it back to `false` otherwise.
```

Comment:
This is not clear to me: it is never introduced the concept of `the result of their respective operation` before this point - how is defined the result of an instruction? As far as I know, the state should depend on the first argument passed to this instruction (but I might be completely wrong)